### PR TITLE
spring security

### PR DESCRIPTION
**Pull request by s-hrynus**
…mplemented jwt token, added all necessary dependencies

### CODE REVIEW

File: src/main/java/mate/academy/security/JwtAuthenticationFilter.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                    userDetails, null, userDetails.getAuthorities());
+            SecurityContextHolder.getContext().setAuthentication(authentication);
+        }
+        filterChain.doFilter(request, response);
+    }
+
+    private String getToken(HttpServletRequest request) {
+        String bearerToken = request.getHeader("Authorization");
+        if (StringUtils.hasText(bearerToken) && bearerToken.startsWith("Bearer ")) {
+            return bearerToken.substring(7);
```
Comment (Ivan95kos) [inline src/main/java/mate/academy/security/JwtAuthenticationFilter.java]: It is better to use the constants instead of Strings and Numbers

**Review by Ivan95kos — APPROVED** [review]
Great job, with a minor comment

File: src/main/java/mate/academy/security/JwtAuthenticationFilter.java (hunk 1 above)
Comment (s-hrynus) [reply src/main/java/mate/academy/security/JwtAuthenticationFilter.java]: totally agree, I'll update it in the next commit

**Review by kozhukhovsky — APPROVED** [review]

**Review by Ivan95kos — CHANGES_REQUESTED** [review]
Merge all previous commits into master branch.
And then create a new PR with commits only for this task, too many files for this task

**Pull request closed without merging**